BibBuddy/app/src/main/java/de/bibbuddy/BookAddManuallyFragment.java

```diff
@@ -0,0 +1,279 @@
+package de.bibbuddy;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+import android.widget.EditText;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The BookAddManuallyFragment is responsible for adding a book manually to a shelf.
+ *
+ * @author Claudia Schönherr
+ */
+public class BookAddManuallyFragment extends Fragment {
+  private final AddBookManuallyListener listener;
+  List<Author> authorList;
+  private boolean validInput;
+  private Long shelfId;
+  private String shelfName;
+  private int redColor;
+  private int greenColor;
+  private Book book;
```

Comment:
Shouldn't this be initialized here (at the latest in onCreateView)?

Reply:
You can't initialize it here